### ISSUE TITLE
Object deep not.toEqual

### ISSUE THREAD

**Issue by soncodi** [alsatian-test/alsatian#448]
According to the docs, `toEqual` on an object uses deep-equals comparison. These tests all pass:

```ts
Expect({ a: 'a' }).toEqual({ a: 'a' });     // ok
Expect({ a: 'a' }).not.toEqual({ a: 'b' }); // ok
Expect({ a: 'a' }).not.toEqual({ a: 'a' }); // incorrect
```
What is the preferred way to achieve `not.toEqual` for objects?

**Comment by jamesrichford** [comment]
Hi @soncodi looks like this is a bug after a brief review of the code it seems that if the not modifier is used the deep equal check will not be done. Looks like we've got a bug, but should have a PR up soon to resolve it :)

**Comment by jamesrichford** [comment]
@soncodi this should now be available in our nightly of you install `alsatian@next`. Any problems let us know but if not this will get published tomorrow in 2.1.1 :)

**Comment by jamesrichford** [comment]
@soncodi published - enjoy :)
